### PR TITLE
Transfer Id should be u64.

### PR DESCRIPTION
**Pull request by nilz3ro**
# About These Changes
* Makes `transfer_id` `u64` throughout the client lib and FFI.
* closes casper-network/casper-node#1577

### CODE REVIEW

File: client/lib/ffi.rs [hunk 1]
```diff
@@ -379,7 +379,7 @@ pub extern "C" fn casper_transfer(
     let node_address = try_unsafe_arg!(node_address);
     let amount = try_unsafe_arg!(amount);
     let maybe_target_account = try_unsafe_arg!(maybe_target_account);
-    let transfer_id = try_unsafe_arg!(transfer_id);
+    // let transfer_id = try_unsafe_arg!(transfer_id);
```
Comment (darthsiroftardis) [inline client/lib/ffi.rs:382]: Leftover comment from the change

File: client/lib/ffi.rs [hunk 2]
```diff
@@ -407,15 +407,15 @@ pub extern "C" fn casper_make_transfer(
...
+    transfer_id: u64,
     deploy_params: *const casper_deploy_params_t,
     payment_params: *const casper_payment_params_t,
     force: bool,
 ) -> casper_error_t {
     let maybe_output_path = try_unsafe_arg!(maybe_output_path);
     let amount = try_unsafe_arg!(amount);
     let maybe_target_account = try_unsafe_arg!(maybe_target_account);
-    let transfer_id = try_unsafe_arg!(transfer_id);
+    // let transfer_id = try_unsafe_arg!(transfer_id);
```
Comment (darthsiroftardis) [inline client/lib/ffi.rs:418]: Leftover comment

**Review by darthsiroftardis — APPROVED** [review]
LGTM, can be merged once stray comments have been removed

File: client/tests/integration_test.rs [hunk 3]
```diff
@@ -833,7 +833,7 @@ mod make_transfer {
     const AMOUNT: &str = "30000000000";
     const TARGET_ACCOUNT: &str =
         "01aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
-    const TRANSFER_ID: &str = "1314";
+    const TRANSFER_ID: u64 = 1314;
```
Comment (goral09) [inline client/tests/integration_test.rs:836]: This means all clients' (if we have any :sweat_smile:) using Rust client in their bigger ecosystem will have to update? If I have scripts that use `casper-client` Rust binary, would they still be able to use a string version?

Comment (Fraser999) [reply client/tests/integration_test.rs:836]: Other clients using our Rust lib (either as a Rust lib or as a C lib) will need to update.  Usage of the binary is unaffected by this change - the binary currently errors out if a user passes something which doesn't parse as a `u64` for the transfer ID arg.

**Review by Fraser999 — CHANGES_REQUESTED** [review]
This PR should also include an update to the client's CHANGELOG.md.

Otherwise LGTM.

**Comment by nilz3ro** [comment]
bors merge

**Review by Fraser999 — APPROVED** [review]

**Comment by nilz3ro** [comment]
bors merge

**Review by EdHastingsCasperLabs — CHANGES_REQUESTED** [review]
hold up on this; we've been requested to change TransferId's to support strings...needs to be discussed prior to merging this

**Pull request closed without merging**